utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/SwaggerBasedJsonSchemaTypeDefinitionExtractor.scala

```diff
@@ -13,21 +13,17 @@ object SwaggerBasedJsonSchemaTypeDefinitionExtractor {
 
   private val mapper: ObjectMapper = ObjectMapperFactory.createJson()
 
-  def swaggerType(schema: Schema): SwaggerTyped = {
+  def swaggerType(schema: Schema, parentSchema: Option[Schema] = None): SwaggerTyped = {
     val deserializedSchema: media.Schema[_] = OpenAPISchemaParser.parseSchema(schema.toString)
-    swaggerType(deserializedSchema)
-  }
-
-  def swaggerType(schema: media.Schema[_]): SwaggerTyped = {
-    val refSchemas = collectSchemaDefs(schema)
-    SwaggerTyped(schema, refSchemas)
+    SwaggerTyped(deserializedSchema, collectSchemaDefs(parentSchema.getOrElse(schema)))
```

Comment:
if `parentSchema` is defined why we don't want to `collectSchemaDefs` also from `schema` and then merge them?

Reply:
well, the idea is that the parent schema contains also refs from schema, but as it costs almost nothing I added also refs from schema as you suggest :+1: